packages/material-ui-lab/src/TreeView/TreeView.js

```diff
@@ -46,12 +46,55 @@ const TreeView = React.forwardRef(function TreeView(props, ref) {
     selected: selectedProp,
     ...other
   } = props;
-  const [tabbable, setTabbable] = React.useState(null);
-  const [focusedNodeId, setFocusedNodeId] = React.useState(null);
   const [descendants, setDescendants] = useDescendantsInit();
 
   const nodeMap = React.useRef({});
 
+  const [{ tabbable, focusedNodeId }, focusDispatch] = React.useReducer(
+    (state, action) => {
+      const { focusedNodeId: previousFocusedNodeId, tabbable: previousTabbable } = state;
+
+      switch (action.type) {
+        case 'blur':
+          return { focusedNodeId: null, tabbable: previousTabbable };
+        case 'collapse': {
+          const parentId = nodeMap.current[previousTabbable]?.parentId;
+          if (previousTabbable && parentId === action.payload) {
+            return { focusedNodeId: previousFocusedNodeId, tabbable: parentId };
```

Comment:
@joshwooding Could you explain what kind of transition this is? This behavior comes from https://github.com/mui-org/material-ui/pull/21720/files#diff-1192977187ec51459a4dd0b09758a929L302.

Shouldn't we also update the `focusedNodeId`?

Reply:
The focus would have been updated with the onClick or arrow navigation. Tbh I spot a lot of issues with the old code. It was supposed to trigger if the current tabbable was a child of the parent but it didn't check at any other depth. Plus it probably never was triggered - I don't think tabbable can get in this state... I would say it's safe to remove this case altogether.

Reply:
Case is currently not covered. Removing it for now.